File: php/features.php
<?php
declare(strict_types=1);

// OpenElevation SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class OpenElevationFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new OpenElevationBaseFeature();
            case "test":
                return new OpenElevationTestFeature();
            default:
                return new OpenElevationBaseFeature();
        }
    }
}

File: py/features.py
# OpenElevation SDK feature factory

from feature.base_feature import OpenElevationBaseFeature
from feature.test_feature import OpenElevationTestFeature


def _make_feature(name):
    features = {
        "base": lambda: OpenElevationBaseFeature(),
        "test": lambda: OpenElevationTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
